src/System.Runtime/tests/System/Text/StringBuilderTests.netcoreapp.cs

```diff
@@ -388,5 +388,34 @@ public static void Equals(StringBuilder sb1, string value, bool expected)
         {
             Assert.Equal(expected, sb1.Equals(value.AsSpan()));
         }
+
+        [Fact]
+        public static void ForEach()
+        {
+            // Test on a variety of lengths, at least up to the point of 9 8K chunks = 72K because this is where
+            // we start using a different technique for creating the ChunkEnumerator.   200 * 500 = 100K which hits this.   
+            for (int i = 0; i < 200; i++)
+            {
+                StringBuilder inBuilder = new StringBuilder();
+                for (int j = 0; j < i; j++)
+                {
+                    // Make some unique strings that are at least 500 bytes long.  
+                    inBuilder.Append(j);
+                    inBuilder.Append("_abcdefghijklmnopqrstuvwxyz01234567890__Abcdefghijklmnopqrstuvwxyz01234567890__ABcdefghijklmnopqrstuvwxyz01_");
+                    inBuilder.Append("_abcdefghijklmnopqrstuvwxyz01234567890__Abcdefghijklmnopqrstuvwxyz01234567890__ABcdefghijklmnopqrstuvwxyz0123_");
+                    inBuilder.Append("_abcdefghijklmnopqrstuvwxyz01234567890__Abcdefghijklmnopqrstuvwxyz01234567890__ABcdefghijklmnopqrstuvwxyz012345_");
+                    inBuilder.Append("_abcdefghijklmnopqrstuvwxyz01234567890__Abcdefghijklmnopqrstuvwxyz01234567890__ABcdefghijklmnopqrstuvwxyz012345678_");
+                    inBuilder.Append("_abcdefghijklmnopqrstuvwxyz01234567890__Abcdefghijklmnopqrstuvwxyz01234567890__ABcdefghijklmnopqrstuvwxyz01234567890_");
+                }
+
+                // Copy the string out (not using StringBuilder).  
+                string outStr = "";
+                foreach (ReadOnlyMemory<char> chunk in inBuilder.EnumerateChunks())
+                    outStr += new string(chunk.Span);
```

Comment:
Do we want any validation for the behavior when the StringBuilder is modified while its chunks are being enumerated?

Reply:
The behavior is undefined when a StringBuilder is modified when enumeration is in progress.     I will put a warning about this in  EnumerateChunks().

I had considered actively throwing if this is detected, but this DOES slow the main Append() path down (not pay for play), and the value here is very low because this foreach API is a pretty rare use case for use by sophisticated users.